pkg/memtable/memtable.go

```diff
@@ -98,17 +98,25 @@ func (mt *Memtable) getOneColl(ctx context.Context, db *mongo.Database, coll, ke
 }
 
 func (mt *Memtable) Put(ctx context.Context, key string, value []byte) (string, error) {
+	rec := &types.Record{
+		Key:       key,
+		Timestamp: mt.clock.Now(),
+		Document:  value,
+	}
+	return mt.PutRecord(ctx, rec)
+}
+
+func (mt *Memtable) PutRecord(ctx context.Context, rec *types.Record) (string, error) {
 	c, err := mt.activeCollection(ctx)
 	if err != nil {
 		return "", err
 	}
```

Comment:
add a check here that rec.Timestamp is nil, and return an error if it's not. it would be confusing for the caller if we just overwrote their timestamp.